cmd/dogstatsd/main.go

```diff
@@ -212,7 +214,29 @@ func runAgent() (mainCtx context.Context, mainCtxCancel context.CancelFunc, err
 	return
 }
 
-func stopAgent(ctx context.Context, cancel context.CancelFunc) {
+// handleSignals handles OS signals, and sends a message on stopCh when an interrupt
+// signal is received.
+func handleSignals(stopCh chan struct{}) {
```

Comment:
There is a logic similar in the trace agent https://github.com/DataDog/datadog-agent/blob/1ffcd3c8a7a776903981892b48f1cadf47302e02/cmd/trace-agent/main.go#L17 Does it worth factorizing this logic?

Reply:
I'd do this in a future PR